mtags/src/main/scala-3/scala/meta/internal/pc/CompilerSearchVisitor.scala

```diff
@@ -28,7 +28,7 @@ class CompilerSearchVisitor(
     owner.isStatic && owner.isPublic
 
   private def isAccessible(sym: Symbol): Boolean = try
-    sym != NoSymbol && sym.isPublic && sym.isStatic ||
+    sym.denot.isAccessibleFrom(sym.paramInfo) ||
```

Comment:
```suggestion
    sym.isAccessibleFrom(site) ||
```
This might be a bit more complicated since we need to have the actual from site (`site: Type`), so the symbol we are currently in. We could forward is to CompilerSearchVisitor and use the typed path for it (probably the first existing type in the path).

Reply:
Yeah, sorry I pushed this change up as it's a bit absurd.

I'll have a deeper think about this.

Reply:
I _think_ what we want is
```scala
sym.isAccessibleFrom(ctx.owner.info)
```

Reply:
Ach, right we already calculate it, good catch. That should be it.

Reply:
I believe this is what we want as the tests are passing in Scala 3.